extra/crc32_armv8_neon/crc32_armv8.c

```diff
@@ -0,0 +1,301 @@
+#include <my_global.h>
+#include <string.h>
+
+
+#if defined(__GNUC__) && defined(__linux__) && defined(HAVE_ARMV8_CRC)
+
+#include <sys/auxv.h>
+#include <asm/hwcap.h>
+
+#ifndef HWCAP_CRC32
+#define HWCAP_CRC32 (1 << 7)
+#endif
+
+unsigned int crc32c_arch64_available(void)
+{
+	unsigned long auxv = getauxval(AT_HWCAP);
+	return (auxv & HWCAP_CRC32) != 0;
```

Comment:
is the crypto intrinsics covered by this same capability test?

Reply:
Sorry what I mean is the ARMV8 CRC CRYPTO runtime guaranteed to be there when the runtime HWCAP_CRC32 is detected. Other references seem to indicate it is.

Reply:
"ARMV8 CRC CRYPTO" is checked in "cmake/crc32_armv8_neon.cmake".
It is just to check that whether the compiler is support the Arm instructions set in building host or not. 
If the Apps is compiled on Host A. It could not guarantee that the Apps that includes these Arm instructions can run in the Host B.
So we should detect the Host B's CPU features first here. 
:)